### DISCUSSION TITLE
10 print turnkey demo

### DISCUSSION THREAD

**Discussion by Whammo** [jkotlinski/durexforth#382] [Show and tell]
Target disk in device 9

```forth
---editor--- \ free the memory
: sidini ( -- )
$ffff $d40e !
$80 $d412 c! ;

: sidrnd ( -- c )
$D41B c@ ;

: doit do sidrnd 1 and if $4d
else $4e then i c! loop ;

: 10print 142 emit sidini
2024 1024 doit
begin
1064 1024 960 move
2024 1984 doit
key? until 64738 sys
key drop page 14 emit ;

include turnkey
' 10print start !
9 device
save-prg 10print
8 device


```

**Comment by lonetech** [comment]
Is this where I post my `maze` exercise code? I've also done the turnkey exercise with `mazedemo`, which can switch between modes. 

	include rnd

	." Usage: ' diag maze or ' ortho maze"
	
	: diag
	 \ 15 rshift 'M' + ;
	 0< 'N' + ;
	
	: maze ( char-xt -- )
	 142 emit \ PETSCII graphics
	 begin
	   rnd over execute emit
	 key? until
	 key drop 14 emit drop ;
	
	create maze$
	 171 c, 177 c, 178 c,
	 179 c,  99 c,  98 c,
	: ortho
	 6 um* nip  \ scale to 0..5
	 maze$ + c@ ;
	
	: mazedemo
	 ['] diag
	 ['] ortho
	 begin
	   swap dup maze
	 again ;
